src/simulators/matrix_product_state/svd.cpp

```diff
@@ -114,33 +124,28 @@ status csvd(cmatrix_t &A, cmatrix_t &U,rvector_t &S,cmatrix_t &V)
   complex_t q = 0;
   // Transpose when m < n
   bool transposed = false;
-  if (m < n)
-    {
-    	transposed = true;
-    	A = AER::Utils::dagger(A);
-	std::swap(m,n);
+  if (m < n) {
+    transposed = true;
+    A = AER::Utils::dagger(A);
+    std::swap(m,n);
+  }
+  cmatrix_t temp_A = A;
+  c[0] = 0;
+  while(true) {
+    k1 = k + 1;
+    z = 0.0;
+    for( i = k; i < m; i++){
+      z = z + norm(A(i,k));
     }
-	cmatrix_t temp_A = A;
-	c[0] = 0;
-	while(true)
-	{
-		k1 = k + 1;
-		z = 0.0;
-		for( i = k; i < m; i++){
-			z = z + norm(A(i,k));
-		}
-		b[k] = 0.0;
-		if ( tol < z )
-		{
+    b[k] = 0.0;
+    if ( tol < z ){
 			z = std::sqrt( z );
 			b[k] = z;
 			w = std::abs( A(k,k) );
-			if (Linalg::almost_equal(static_cast<long double>(w), 
```

Comment:
For these sort of comparisons you can specify the type by the function template parameters instead of casting the argument `Linalg::almost_equal<long double>(a, b, max_diff)`

Reply:
I started checking this, and while doing so, I noticed another problem. So I will need some more time to work on this PR. I am adding back the WIP.